tests/FunctionsTest.php

```diff
@@ -135,4 +169,53 @@ public function provideMapReduceOutValues()
             [ ['replace' => 'collectionName'], false ],
         ];
     }
+
+    /**
+     * @dataProvider provideTypeMapValues
+     */
+    public function testCreateFieldPathTypeMap(array $expected, array $typeMap, $fieldPath = 'field')
+    {
+        $this->assertEquals($expected, \MongoDB\create_field_path_type_map($typeMap, $fieldPath));
+    }
+
+    public function provideTypeMapValues()
+    {
+        return [
+            'No root type' => [
+                ['document' => 'array', 'root' => 'object'],
+                ['document' => 'array'],
+            ],
+            'No field path' => [
+                ['root' => 'object', 'fieldPaths' => ['field' => 'array']],
+                ['root' => 'array'],
+            ],
+            'Field path exists' => [
+                ['root' => 'object', 'fieldPaths' => ['field' => 'object']],
+                ['root' => 'array', 'fieldPaths' => ['field' => 'object']],
```

Comment:
Related to my earlier comments on `create_field_path_type_map()`, I think this is where we'd expect:

```
[
    'root' => 'object',
    'fieldPaths' => [
        'field' => 'array',
        'field.field' => 'object',
    ],
]
```

At least I believe that's how this would have worked previously with TypeMapArrayIterator. It's quite possible we never had tests combining `fieldPaths` with TypeMapArrayIterator, since the class predated that extension feature.